components/tidb_query_vec_executors/src/table_scan_executor.rs

```diff
@@ -291,21 +299,46 @@ impl ScanExecutorImpl for TableScanExecutorImpl {
     ) -> Result<()> {
         use tidb_query_datatype::codec::{datum, table};
 
-        check_record_key(&key)?;
         let columns_len = self.schema.len();
         let mut decoded_columns = 0;
 
+        // `handle_indices` and `primary_column_ids` mutexes.
+        assert!(self.handle_indices.is_empty() || self.primary_column_ids.is_empty());
```

Comment:
return error is more friendly

Reply:
Why we don't do this check when `BatchTableScanExecutor` is constructed?